### ISSUE TITLE
`ClassCastException` in `kotlinx-serialization-runtime-js`

### ISSUE THREAD

**Issue by flaghacker** [Kotlin/kotlinx.serialization#26]
When trying run the following code in Kotlin-JS:

```
import kotlinx.serialization.json.JSON

data class Foo(val foo: String)

fun main(args: Array<String>) {
    println("Hello World!")
    println(JSON.stringify(Foo("Hey")))
}
```
I get this error:
```
kotlin.js:3289 Hello World!
kotlin.js:3446 Uncaught ClassCastException "Illegal cast"
"ClassCastException: Illegal cast
    at Object.throwCCE (file:///.../web/lib/kotlin.js:3446:13)
    at serializer (file:///.../web/lib/kotlinx-serialization-runtime-js.js:5499:91)
    at main (file:///.../web/main.js:42:37)
    at file:///D:/.../web/main.js:47:3
    at file:///D:/.../web/main.js:50:2"
```

(error message and stacktrace slightly edited for readability)

I tested this in both Firefox and Chrome. A repository to reproduce this bug: https://github.com/flaghacker/KotlinSerialBug, exact instructions in `README.md`.

**Comment by sandwwraith** [comment]
Seems that you've forgot to `apply plugin: 'kotlinx-serialization'` in your `build.gradle`
Also, `Foo` class should have `@Serializable` annotation

**Comment by flaghacker** [comment]
That worked, thanks! Would it be possible to make the exception message more clear?

**Comment by sandwwraith** [comment]
It was fixed in version `0.1.1`: https://github.com/Kotlin/kotlinx.serialization/commit/aed36a1d8439457b54c7aa5eead147ee97f1cc48#diff-5276599350027e62e0a0883713ed8580R23

Yet you're using `0.1`

**Comment by flaghacker** [comment]
Thanks, that works.
